plugins/modules/lan_automation_workflow_manager.py

```diff
@@ -0,0 +1,2677 @@
+                self.status = "success"
+                self.set_operation_result("success", True, self.msg, "INFO")
+                break
+
+            self.log("Current progress for task ID {}: {}".format(task_id, task_details.get('progress')), "DEBUG")
+            elapsed_time = time.time() - start_time
+            if int(elapsed_time) % 300 == 0:
+                logs = self.collect_logs()
+                if logs:
+                    self.log("Collected logs: {}".format(logs), "INFO")
+                    self.msg = "LAN Automation Session Logs: {}".format(logs)
+
+            if pnp_authorization and pending_authorization:
+                if remaining_auth_devices:
+                    self.log(
+                        "Authorizing devices in PnP with serial numbers provided: {}".format(remaining_auth_devices),
+                        "DEBUG")
+                    authorized_devices = self.authorize_devices(remaining_auth_devices)
+                    self.log("Authorized devices: {}".format(', '.join(authorized_devices)), "INFO")
+                    remaining_auth_devices = [device for device in remaining_auth_devices if
+                                              device not in authorized_devices]
+
+                    if not authorized_devices:
+                        self.log(
+                            "Some devices from {} were not authorized as their state is not Pending Authorization. "
+                            "Please check the logs for details regarding the state of the devices and ensure the "
+                            "serial numbers are correct. We will keep checking the state and attempt authorizing the "
+                            "devices again. If device authorization is done manually or the checkbox to enable "
+                            "authorization on the device is not checked on Catalyst Center, consider stopping "
+                            "LAN Automation session.".format(remaining_auth_devices), "DEBUG")
+
+                    if remaining_auth_devices:
+                        self.log("Devices still pending authorization: {}".format(', '.join(remaining_auth_devices)),
+                                 "INFO")
+                    else:
+                        pending_authorization = False
+                        self.log("All devices have been successfully authorized.", "INFO")
+                else:
+                    pending_authorization = False
+                    self.msg = ("PnP authorization is set to True, but no devices were given for authorization or "
+                                "discovery. Please authorize devices manually in the PnP page in Catalyst Center.")
+                    self.set_operation_result("failed", False, self.msg, "ERROR")
+
+            self.log("Waiting for 30 seconds...")
+            time.sleep(self.params.get("dnac_task_poll_interval", 30))
+
+        if self.status != "success":
```

Comment:
elapsed_time = time.time() - start_time

        "LAN automation did not complete within the expected time of {:.2f} seconds. Consider stopping the "